src/Middleware/ResponseCompression/src/ResponseCompressionProvider.cs

```diff
@@ -170,6 +171,16 @@ public virtual ICompressionProvider GetCompressionProvider(HttpContext context)
         /// <inheritdoc />
         public virtual bool ShouldCompressResponse(HttpContext context)
         {
+            var httpsMode = context.Features.Get<IHttpsCompressionFeature>()?.Mode ?? HttpsCompressionMode.Default;
+
+            if (context.Request.IsHttps
+                && (httpsMode == HttpsCompressionMode.DoNotCompress
+                    || !(_enableForHttps || httpsMode == HttpsCompressionMode.Compress)))
```

Comment:
Might want to add a comment to list the cases so it's easier to understand since the override logic isn't exactly intuitive.

Reply:
Don't you need to check if httpsMode is HttpsCompressionMode.Default? 

Make sure you add a test too.

Reply:
No, default falls back to _enableForHttps.